pkg/lightning/backend/backend.go

```diff
@@ -197,6 +196,14 @@ type AbstractBackend interface {
 
 	// LocalWriter obtains a thread-local EngineWriter for writing rows into the given engine.
 	LocalWriter(ctx context.Context, cfg *LocalWriterConfig, engineUUID uuid.UUID) (EngineWriter, error)
+
+	// CollectLocalDuplicateRows collect duplicate keys from local db. We will store the duplicate keys which
+	//  may be repeated with other keys in local data source.
+	CollectLocalDuplicateRows(ctx context.Context, tbl table.Table) error
+
+	// CollectLocalDuplicateRows collect duplicate keys from remote TiKV storage. This keys may be duplicate with
+	//  the data import by other lightning.
+	CollectRemoteDuplicateRows(ctx context.Context, tbl table.Table) error
```

Comment:
We probably need to update the name and comment of this method, since `CollectRemoteDuplicateRows` also collect duplicate keys from local db.